webapp/src/Service/ConfigurationService.php

```diff
@@ -0,0 +1,211 @@
+<?php declare(strict_types=1);
+
+namespace App\Service;
+
+use App\Config\Loader\YamlConfigLoader;
+use App\Entity\Configuration;
+use Doctrine\ORM\EntityManagerInterface;
+use Exception;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Config\ConfigCacheFactoryInterface;
+use Symfony\Component\Config\ConfigCacheInterface;
+use Symfony\Component\Config\FileLocator;
+use Symfony\Component\Config\Resource\FileResource;
+
+/**
+ * Class ConfigurationService
+ *
+ * @package App\Service
+ */
+class ConfigurationService
+{
+    /**
+     * @var EntityManagerInterface
+     */
+    protected $em;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @var ConfigCacheFactoryInterface
+     */
+    protected $configCache;
+
+    /**
+     * @var bool
+     */
+    protected $debug;
+
+    /**
+     * @var string
+     */
+    protected $cacheDir;
+
+    /**
+     * @var string
+     */
+    protected $etcDir;
+
+    /**
+     * @var array
+     */
+    protected $dbConfigCache = [];
+
+    /**
+     * ConfigurationService constructor.
+     *
+     * @param EntityManagerInterface      $em
+     * @param LoggerInterface             $logger
+     * @param ConfigCacheFactoryInterface $configCache
+     * @param bool                        $debug
+     * @param string                      $cacheDir
+     * @param string                      $etcDir
+     */
+    public function __construct(
+        EntityManagerInterface $em,
+        LoggerInterface $logger,
+        ConfigCacheFactoryInterface $configCache,
+        bool $debug,
+        string $cacheDir,
+        string $etcDir
+    ) {
+        $this->em          = $em;
+        $this->logger      = $logger;
+        $this->configCache = $configCache;
+        $this->debug       = $debug;
+        $this->cacheDir    = $cacheDir;
+        $this->etcDir      = $etcDir;
+    }
+
+    /**
+     * Get the value for the given configuration name
+     *
+     * @param string $name         The config name to get the value of
+     * @param mixed  $default      The default value to use if the config
+     *                             option and spec are not found
+     * @param bool   $onlyIfPublic Only return the value if the config is
+     *                             public
+     *
+     * @return mixed The configuration value
+     * @throws Exception If the config can't be found and not default is
+     *                   supplied
+     */
```

Comment:
This is what the old behaviour was. I guess it makes sense to not have this anymore, since adding config stuff is easier now. I will remove this functionality.

Reply:
I know this is a bit late, but...
It looks like the doc still refers to the old behaviour: "<span>@</span>throws Exception If the config can't be found and not default supplied".
A consequence of the new behaviour is that if an entry is ever removed from the YAML file, migrations can't use ConfigurationService::get to read the value from the DB (but they can use the Configuration repository directly). I ran into this with #678.

Reply:
Aha you are right. I think it is fine that you should not be able to read nonexisting configuration values, but we should indeed update the comment and such.